src/components/executive-education-2u/UserEnrollmentForm.test.jsx

```diff
@@ -397,4 +411,43 @@ describe('UserEnrollmentForm', () => {
     // disabled after submitting
     expect(screen.getByText('Registration confirmed').closest('button')).toHaveAttribute('aria-disabled', 'true');
   });
+
+  it('handles duplicate order with form submission', async () => {
+    const mockError = new Error('duplicate order');
+    Date.now = jest.fn(() => new Date().valueOf());
+    const mockFormSubmissionValue = { message: 'duplicate order' };
+
+    jest.mock('./data', () => ({
+      ...jest.requireActual('./data'),
+      isDuplicateExternalCourseOrder: jest.fn(() => true),
+    }));
```

Comment:
nit: could likely simplify without the mock here, given `isDuplicateExternalCourseOrder` is a fairly simple function and its return value could be controlled by the `externalCourseFormSubmissionError` passed via `CourseContext`.